### PR TITLE
Port life of dist txn doc to 2.1 docs

### PR DESCRIPTION
**Pull request by jseldess**

### CODE REVIEW

File: v2.1/architecture/life-of-a-distributed-transaction.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+Once the node that contains the leaseholder of the range receives the `BatchRequest`, it begins processing it, and progresses onto checking the timestamp cache.
+
+### Timestamp cache
+
+The timestamp cache tracks the highest timestamp (i.e., most recent) for any read operation that a given range has served.
+
+Each write operation in a `BatchRequest` checks its own timestamp versus the timestamp cache to ensure that the write operation has a higher timestamp; this guarantees that history is never rewritten and you can trust that reads always served the most recent data. It's one of the crucial mechanisms CockroachDB uses to ensure serializability. If a write operation fails this check, it must be restarted at a timestamp higher than the timestamp cache's value.
+
+### Latch manager
```
[the remark this answers is not in the document]
Comment (jseldess) [reply v2.1/architecture/life-of-a-distributed-transaction.md]: Merging now so this content is available in our stable docs, but we can make updates once we hear back from @nvanbenschoten.

[the remark this answers is not in the document]
Comment (nvanbenschoten) [reply v2.1/architecture/life-of-a-distributed-transaction.md]: It was. It's actually no longer called the CommandQueue now though, as of a few weeks ago. We should switch references of it over to the "latch manager" in the docs, though there's no particular urgency in doing so.

[the remark this answers is not in the document]
Comment (jseldess) [reply v2.1/architecture/life-of-a-distributed-transaction.md]: @nvanbenschoten, this name change happened just in 2.2, or also in 2.1?

[the remark this answers is not in the document]
Comment (nvanbenschoten) [reply v2.1/architecture/life-of-a-distributed-transaction.md]: Only in 2.2.

**Pull request merged**